### PR TITLE
Factor out grpc over vsock; speed up goinit, use fast kernel image.

### PR DESCRIPTION
**Pull request by tylerwilliams**
See title.

### CODE REVIEW

File: enterprise/server/cmd/goinit/main.go [hunk 1]
```diff
@@ -35,6 +34,7 @@ var (
 	path      = flag.String("path", "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "The path to use when executing cmd")
 	port      = flag.Uint("port", vsock.DefaultPort, "The vsock port number to listen on")
 	debugMode = flag.Bool("debug_mode", false, "If true, attempt to set root pw and start getty.")
+	logLevel  = flag.String("logLevel", "info", "The loglevel to emit logs at")
```
Comment (bduffany) [inline enterprise/server/cmd/goinit/main.go:37]: nit: rename the cmd line flag to `log_level` (for case consistency w/ other flags)

File: enterprise/server/util/vsock/vsock.go [hunk 2]
```diff
@@ -132,3 +136,42 @@ func DialHostToGuest(ctx context.Context, socketPath string, port uint32) (net.C
 	}
 	return conn, nil
 }
+
+// SimpleGRPCDial internally calls DialHostToGuest and then sets up a gRPC
```
Comment (bduffany) [inline enterprise/server/util/vsock/vsock.go:140]: could `DialHostToGuest` be made private now or is it still used elsewhere?

**Review by bduffany — APPROVED** [review]

**Comment by bduffany** [comment]
Is the "fast kernel image" part in another PR or am I missing something?

File: enterprise/server/cmd/goinit/main.go (hunk 1 above)
Comment (tylerwilliams) [reply enterprise/server/cmd/goinit/main.go]: whoops, good catch. Done.

File: enterprise/server/util/vsock/vsock.go (hunk 2 above)
Comment (tylerwilliams) [reply enterprise/server/util/vsock/vsock.go:140]: Done.

**Review by tylerwilliams — COMMENTED** [review]
new kernel is in enterprise/vmsupport/bin/vmlinux

**Pull request merged**